### PR TITLE
Implement sort list for applicant

### PR DESCRIPTION
**Pull request by khoahre123**
The list command now accepts optional arguments `SORT_ARGUMENT` beside the `filter` function. The syntax for sort command is list -X s/SORT_ARGUMENT.

### CODE REVIEW

**Comment by khoahre123** [comment]
Able to sort the list into position, but can not display into UI. Need to change UI

File: src/main/java/seedu/address/model/interview/Interview.java [hunk 1]
```diff
@@ -32,10 +32,11 @@ public Interview(Applicant applicant, LocalDateTime date, Position position) {
     }
 
     /**
+     * Create Interview object when loading from database
      * Every field must be present and not null.
      */
     public Interview(Applicant applicant, LocalDateTime date, Position position, Status status) {
-        requireAllNonNull(applicant, date);
+        requireAllNonNull(applicant, date, status);
```
Comment (goalfix) [inline src/main/java/seedu/address/model/interview/Interview.java:39]: Position should not be null too

**Review by goalfix — CHANGES_REQUESTED** [review]
Some issues with code quality raised.  Good effort regardless!

**Review by goalfix — APPROVED** [review]
LGTM!

**Pull request merged**